fvm/evm/types/block.go

```diff
@@ -141,3 +123,145 @@ var GenesisBlock = &Block{
 }
 
 var GenesisBlockHash, _ = GenesisBlock.Hash()
+
+// Below block type section, defines earlier block types,
+// this is being used to decode blocks that were stored
+// before block type changes. It allows us to still decode
+// a block that would otherwise be invalid if decoded into
+// latest version of the above Block type.
```

Comment:
Do we need to keep these forever or just until previewnet is shutdown?

Can you include a note about that in this comment.